pkg/block/mem/adapter.go

```diff
@@ -63,11 +64,11 @@ type Adapter struct {
 }
 
 // FetchStore Returns mem store if already initialized, otherwise creates one
-func FetchStore(opts ...func(a *Adapter)) *Adapter {
-	if memStore != nil {
+func FetchStore(params params.Mem) *Adapter {
```

Comment:
Why would I call `FetchStore` with `params.ReuseStore=false`? I think it makes this parameter or this function redundant.